src/api/v2/ResourceGroupsExtensionManager.ts

```diff
@@ -42,13 +43,13 @@ function getInactiveExtensions(): vscode.Extension<unknown>[] {
 }
 
 export class ResourceGroupsExtensionManager {
-    async activateApplicationResourceBranchDataProvider(type: string): Promise<void> {
-        type = type.toLowerCase();
+    async activateApplicationResourceBranchDataProvider(type: AzExtResourceType): Promise<void> {
+        const normalizedType = type.toLowerCase();
```

Comment:
This step isn't desirable. These are strings but they should be treated as constants.

Reply:
Within the context of *this* extension, sure.  This is just an attempt at being reasonably permissive with respect to how other extensions may register their BDPs in *their* `package.json`.  We *could* be super strict about matching the exact case of the resource type, but do we need to be?

Reply:
If we aren't strict, I can see us quickly finding our way back to always having to do `toLowerCase()` all over the place again.

Reply:
I don't think normalizing at the "human" boundaries is a bad thing, or will necessarily result in `toLowerCase()` further infiltrating the codebase.  That said, it's something that can be added if/when we find those humans insisting on their own arbitrary casing and I just don't feel strongly about it either way, so I've reverted it.